persistence/src/main/java/com/epam/esm/persistence/dao/impl/CertificateDAOImpl.java

```diff
@@ -0,0 +1,112 @@
+
+        template.update(connection -> {
+            PreparedStatement ps = connection
+                    .prepareStatement(CertificateQueries.INSERT_CERTIFICATE.getValue(),
+                            Statement.RETURN_GENERATED_KEYS);
+            ps.setString(1, certificate.getName());
+            ps.setString(2, certificate.getDescription());
+            ps.setBigDecimal(3, certificate.getPrice());
+            ps.setInt(4, certificate.getDuration());
+            ps.setDate(5, Date.valueOf(certificate.getCreateDate()));
+            ps.setDate(6, Date.valueOf(certificate.getLastUpdateDate()));
+            return ps;
+        }, keyHolder);
+        if (keyHolder.getKey() == null) {
+            throw new DAOSQLException("empty keyholder");
+        }
+        certificate.setId(keyHolder.getKey().intValue());
+        return certificate;
+    }
+
+    @Override
+    public Certificate read(int id) throws DAOSQLException {
+        return template.query(CertificateQueries.SELECT_FROM_CERTIFICATE.getValue()
+                        .concat(CertificateQueries.WHERE_ID.getValue()),
+                certificateMapper, id).stream().findFirst().get();
+    }
+
+    @Override
+    public void update(Certificate certificate) throws DAOSQLException {
+        template.update(CertificateQueries.UPDATE_CERTIFICATE.getValue(), certificate.getName(),
+                certificate.getDescription(),
+                certificate.getPrice(),
+                certificate.getDuration(),
+                Date.valueOf(certificate.getCreateDate()),
+                Date.valueOf(certificate.getLastUpdateDate()),
+                certificate.getId());
+    }
+
+    @Override
+    public void delete(int id) throws DAOSQLException {
+        template.update(CertificateQueries.DELETE_CERTIFICATE.getValue(), id);
+    }
+
+    @Override
+    public List<Certificate> findAll() throws DAOSQLException {
+        return template.query(CertificateQueries.SELECT_FROM_CERTIFICATE.getValue(), certificateMapper);
+    }
+
+    @Override
+    public void addCertificateTag(int certificateId, int tagId) {
+        template.update(CertificateQueries.ADD_CERTIFICATE_TAG.getValue(),
+                certificateId, tagId);
+    }
+
+    @Override
+    public void deleteCertificateTag(int certificateId, int tagId) {
+        template.update(CertificateQueries.DELETE_CERTIFICATE_TAG.getValue(),
+                certificateId, tagId);
+    }
+
+    @Override
+    public List<Certificate> findBy(EntityFinder<Certificate> finder) throws DAOSQLException {
+        return template.queryForStream(CertificateQueries.SELECT_FROM_CERTIFICATE_TAG.getValue()
+                        .concat(finder.getQuery()),
+                certificateMapper).distinct().collect(Collectors.toList());
+    }
+
+    public boolean isTagCertificateTied(int certificateId, int tagId) {
+        if (template.queryForObject(CertificateQueries.COUNT_CERTIFICATE_TAG.getValue(),
```

Comment:
you can remove if...else and simply return:
int count = template.queryForObject(...);
return count > 0;

also could you please double check that queryForObject with this parameters is not deprecated?

Reply:
It is deprecated, but  queryForObject(String sql, Class<T> requiredType, @Nullable Object... args) didn't work, did not substitute "?" with args. 
Use instead T queryForObject(String sql,
                                      Object[] args,
                                      int[] argTypes,
                                      Class<T> requiredType)